csrc/ir/internal_nodes.h

```diff
@@ -3442,4 +3442,76 @@ class CutlassNvfp4GroupedMmaOp : public Expr {
   }
 };
 
+//! NOTE -- [ GroupedBlockScalingFactorLayoutOp ]
+//!
+//! This operation performs a layout change on the input, it's currently used
+//! for block scaling factor accompanying narrow precision inputs.
+//!
+//! 1. This can be viewed as a point-wise operation, where output loop domain
+//! would match the input logical domain.
+//! 2. Because of the potential padding/swizzle, the logical domain of the
+//! output does not map to input. We don't rely on codegen for indexing, so we
+//! don't care about mapping the logical/allocation of output to anything else.
+//! Indexing will be done in runtime function, utilizing `expert_offsets` and
+//! `sf_offsets`.
+//! 3. Output has a root domain, which is identical to its loop domain. We add
+//! this so we can map it to input.
+class GroupedBlockScalingFactorLayoutOp : public Expr {
```

Comment:
Maybe something like `PreprocessGroupedMatmulInput`?

Reply:
`PreprocessGroupedMatmulInputSf` added `Sf` for scaling factor. Since this doesn't apply to the quantized input.